File: Zisimopoulos_Valencia/Software Construction/QLS/src/uva/sc/qls/logic/Section.java
package uva.sc.qls.logic;

import uva.sc.qls.ast.IQLSNode;
import uva.sc.qls.ast.IQLSNodeVisitor;

public class Section implements IQLSNode {

	String		str;
	SectionBody	sectionBody;

	public Section(String str, SectionBody sectionBody) {
		this.str = str;
		this.sectionBody = sectionBody;
	}

	public SectionBody getSectionBody() {
		return sectionBody;
	}

	public <T> T accept(IQLSNodeVisitor<T> visitor) {
		return visitor.visit(this);
	}

	public String toString() {
		String result = "[Section]: " + str + "\n\t";
		result += sectionBody.toString();
		return result;
	}

}

File: CarlaErik/UvA.SoftCon.Questionnaire/QLS/AST/Model/Section.cs
using System.Collections.Generic;
using UvA.SoftCon.Questionnaire.Common.AST.Model;

namespace UvA.SoftCon.Questionnaire.QLS.AST.Model
{
    public class Section : QLSNode
    {
        public string Title
        {
            get;
            private set;
        }

        public IEnumerable<QuestionReference> QuestionReferences
        {
            get;
            private set;

        }
        public IEnumerable<DefaultStyle> DefaultStyles
        {
            get;
            private set;
        }

        internal Section(string title, IEnumerable<QuestionReference> questionReferences, IEnumerable<DefaultStyle> defaultStyles, TextPosition position)
            : base(position) 
        {
            Title = title;
            QuestionReferences = questionReferences;
            DefaultStyles = defaultStyles;
        }

        public override T Accept<T>(IStyleSheetVisitor<T> visitor)
        {
            return visitor.VisitSection(this);
        }
    }
}
